src/graphql/data/TopTokens.ts

```diff
@@ -112,45 +114,42 @@ function useFilteredTokens(tokens: NonNullable<TopTokens100Query['response']['to
 
 // Number of items to render in each fetch in infinite scroll.
 export const PAGE_SIZE = 20
-
-export type TopToken = NonNullable<NonNullable<TopTokens100Query['response']>['topTokens']>[number]
 export type SparklineMap = { [key: string]: PricePoint[] | undefined }
+export type TopToken = NonNullable<NonNullable<TopTokens100Query>['topTokens']>[number]
+
 interface UseTopTokensReturnValue {
   tokens: TopToken[] | undefined
+  loadingTokens: boolean
   sparklines: SparklineMap
 }
 
 export function useTopTokens(chain: Chain): UseTopTokensReturnValue {
   const chainId = CHAIN_NAME_TO_CHAIN_ID[chain]
   const duration = toHistoryDuration(useAtomValue(filterTimeAtom))
 
-  const environment = useRelayEnvironment()
-  const [sparklines, setSparklines] = useState<SparklineMap>({})
-  useEffect(() => {
-    const subscription = fetchQuery<TopTokensSparklineQuery>(environment, tokenSparklineQuery, { duration, chain })
-      .map((data) => ({
-        topTokens: data.topTokens?.map((token) => unwrapToken(chainId, token)),
-      }))
-      .subscribe({
-        next(data) {
-          const map: SparklineMap = {}
-          data.topTokens?.forEach(
-            (current) =>
-              current?.address && (map[current.address] = current?.market?.priceHistory?.filter(isPricePoint))
-          )
-          setSparklines(map)
-        },
-      })
-    return () => subscription.unsubscribe()
-  }, [chain, chainId, duration, environment])
+  const { data: sparklineData } = useTopTokensSparklineQuery({
+    variables: { duration, chain },
+    fetchPolicy: 'cache-and-network',
+  })
 
-  useEffect(() => {
-    setSparklines({})
-  }, [duration])
+  const sparklines = useMemo(() => {
+    const unwrappedTokens = sparklineData?.topTokens?.map((topToken) => unwrapToken(chainId, topToken))
```

Comment:
would it be cleaner to have chain be from `topToken` and then `sparklineData?.topTokens?.map(unwrapToken)`

Reply:
See comment on scope creep.